src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -5329,7 +5329,7 @@ internal static SessionStateAliasEntry[] BuiltInAliases
         internal const string DefaultMoreFunctionText = @"
 param([string[]]$paths)
 # Nano needs to use Unicode, but Windows and Linux need the default
-$OutputEncoding = if ($IsWindows -and $IsCoreCLR) {
+$OutputEncoding = if ([System.Management.Automation.Platform]::IsNanoServer) {
```

Comment:
You also need to check `[System.Management.Automation.Platform]::IsIoT`. IoT is very similar to NanoServer.